GameServerCore/Domain/IBuffManager.cs

```diff
@@ -0,0 +1,156 @@
+﻿using GameServerCore.Domain.GameObjects;
+using Priority_Queue;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace GameServerCore.Domain
+{
+    public interface IBuffManager
+    {
+
+        /// <summary>
+        /// get the number of buffs
+        /// </summary>
+        /// <returns>buff count</returns>
+        int Count();
+
+        /// <summary>
+        /// add a buff
+        /// </summary>
+        /// <param name="buff">buff to add</param>
+        void Add(IBuff buff);
+
+        /// <summary>
+        /// add multiple buffs
+        /// </summary>
+        /// <param name="buffs">list of buffs</param>
+        void Add(List<IBuff> buffs);
+
+        /// <summary>
+        /// get all buffs
+        /// </summary>
+        /// <returns>list of all buffs</returns>
+        IEnumerable<IBuff> Get();
+
+        /// <summary>
+        /// get a buff by name
+        /// </summary>
+        /// <param name="buffName">name of buff</param>
+        /// <returns>buff</returns>
+        IBuff Get(string buffName);
+
+        /// <summary>
+        /// get a buff by filter
+        /// </summary>
+        /// <param name="filter">filter for buff</param>
+        /// <returns>buff</returns>
+        IBuff Get(Func<IBuff, bool> filter);
+
+        /// <summary>
+        /// get all buffs with name
+        /// </summary>
+        /// <param name="buffName">buff name</param>
+        /// <returns>buffs</returns>
+        IEnumerable<IBuff> GetAll(string buffName);
+
+        /// <summary>
+        /// get a filtered list of buffs
+        /// </summary>
+        /// <param name="filter">filter</param>
+        /// <returns>buffs</returns>
+        IEnumerable<IBuff> GetAll(Func<IBuff, bool> filter);
+
+        /// <summary>
+        /// get the buff priority queue
+        /// </summary>
+        /// <returns>buff priority queue</returns>
+        SimplePriorityQueue<IBuff, float> GetQueue();
```

Comment:
I think that it is incorrect to expose implementation here. I don't see any need of using queue. The `BuffManger` can have a function like `GetNextElapsedBuff` by inspecting the buff in the top of the queue and then just update all the buffs (and remove those with time that is before/equal to this elapsed time).